packages/fuels-core/src/types/wrappers/transaction.rs

```diff
@@ -136,10 +182,15 @@ impl From<TransactionType> for FuelTransaction {
         match value {
             TransactionType::Script(tx) => tx.into(),
             TransactionType::Create(tx) => tx.into(),
+            TransactionType::Mint(tx) => tx.into(),
         }
     }
 }
 
+fn unsupported_mint_operation(operation: &str) -> ! {
+    unimplemented!("Calling `{operation}` on Mint transactions is unsupported!");
```

Comment:
Maybe it'd be better to just return an error instead of panicking? Imo we shouldn't have any public apis exposed to users that allow them to cause a panic.

Reply:
I started a discussion about not implementing this trait at all. If that for some reason isn't accepted, I'll change the trait so that methods that would have passed normally on a Script or Create will fail in the case of a Mint transaction.